src/databricks/labs/ucx/azure/access.py

```diff
@@ -93,29 +93,29 @@ def save_spn_permissions(self) -> str | None:
     def _update_cluster_policy_definition(
         self,
         policy_definition: str,
-        storage_accounts: list[AzureResource],
+        storage_accounts: list[StorageAccount],
         uber_principal: PrincipalSecret,
         inventory_database: str,
     ) -> str:
         policy_dict = json.loads(policy_definition)
         tenant_id = self._azurerm.tenant_id()
         endpoint = f"https://login.microsoftonline.com/{tenant_id}/oauth2/token"
         for storage in storage_accounts:
-            policy_dict[
-                f"spark_conf.fs.azure.account.oauth2.client.id.{storage.storage_account}.dfs.core.windows.net"
-            ] = self._policy_config(uber_principal.client.client_id)
-            policy_dict[
-                f"spark_conf.fs.azure.account.oauth.provider.type.{storage.storage_account}.dfs.core.windows.net"
-            ] = self._policy_config("org.apache.hadoop.fs.azurebfs.oauth2.ClientCredsTokenProvider")
-            policy_dict[
-                f"spark_conf.fs.azure.account.oauth2.client.endpoint.{storage.storage_account}.dfs.core.windows.net"
-            ] = self._policy_config(endpoint)
-            policy_dict[f"spark_conf.fs.azure.account.auth.type.{storage.storage_account}.dfs.core.windows.net"] = (
+            policy_dict[f"spark_conf.fs.azure.account.oauth2.client.id.{storage.name}.dfs.core.windows.net"] = (
```

Comment:
diff in this method doesn't seem to be strictly necessary. can we avoid it in this PR?

Reply:
I needed to update the `AzureResource` -> `StorageAccount` to have a location for creating the access connector. That lead to this change